login/service_blackbox_test.go

```diff
@@ -0,0 +1,198 @@
+package login_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/app"
+	. "github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+)
+
+var db *gorm.DB
+var loginService Service
+
+func TestMain(m *testing.M) {
+	if _, c := os.LookupEnv(resource.Database); c == false {
+		fmt.Printf(resource.StSkipReasonNotSet+"\n", resource.Database)
+		return
```

Comment:
This will skip all tests in the 'login' package if no Database is available, regardless of 'the other' tests being defined as Unit or not. See https://github.com/almighty/almighty-core/blob/master/workitem_whitebox_test.go#L30 for workaround